File: Payhub - SDK/php_SDK/com/payhub/ws/model/RefundResponse.php
<?php

/**
 * Created by PhpStorm.
 * User: agustin
 * Date: 28/07/2015
 * Time: 09:19
 */
class RefundResponse
{
    public $saleTransactionId;

    public $refundTransactionId;

    public $token;

    public $approvalCode;

    public $processedDateTime;

    public $avsResultCode;

    public $verificationResultCode;

    public $batchId;

    public $responseCode;

    public $responseText;

    public $cisNote;

    public $riskStatusResponseText;

    public $riskStatusRespondeCode;

    public $refundDateTime;

    public $billingReferences;

    public $customerReference;

    /**
     * @return mixed
     */
    public function getSaleTransactionId()
    {
        return $this->saleTransactionId;
    }

    /**
     * @param mixed $saleTransactionId
     */
    public function setSaleTransactionId($saleTransactionId)
    {
        $this->saleTransactionId = $saleTransactionId;
    }

    /**
     * @return mixed
     */
    public function getRefundTransactionId()
    {
        return $this->refundTransactionId;
    }

    /**
     * @param mixed $refundTransactionId
     */
    public function setRefundTransactionId($refundTransactionId)
    {
        $this->refundTransactionId = $refundTransactionId;
    }

    /**
     * @return mixed
     */
    public function getToken()
    {
        return $this->token;
    }

    /**
     * @param mixed $token
     */
    public function setToken($token)
    {
        $this->token = $token;
    }

    /**
     * @return mixed
     */
    public function getApprovalCode()
    {
        return $this->approvalCode;
    }

    /**
     * @param mixed $approvalCode
     */
    public function setApprovalCode($approvalCode)
    {
        $this->approvalCode = $approvalCode;
    }

    /**
     * @return mixed
     */
    public function getProcessedDateTime()
    {
        return $this->processedDateTime;
    }

    /**
     * @param mixed $processedDateTime
     */
    public function setProcessedDateTime($processedDateTime)
    {
        $this->processedDateTime = $processedDateTime;
    }

    /**
     * @return mixed
     */
    public function getAvsResultCode()
    {
        return $this->avsResultCode;
    }

    /**
     * @param mixed $avsResultCode
     */
    public function setAvsResultCode($avsResultCode)
    {
        $this->avsResultCode = $avsResultCode;
    }

    /**
     * @return mixed
     */
    public function getVerificationResultCode()
    {
        return $this->verificationResultCode;
    }

    /**
     * @param mixed $verificationResultCode
     */
    public function setVerificationResultCode($verificationResultCode)
    {
        $this->verificationResultCode = $verificationResultCode;
    }

    /**
     * @return mixed
     */
    public function getBatchId()
    {
        return $this->batchId;
    }

    /**
     * @param mixed $batchId
     */
    public function setBatchId($batchId)
    {
        $this->batchId = $batchId;
    }

    /**
     * @return mixed
     */
    public function getResponseCode()
    {
        return $this->responseCode;
    }

    /**
     * @param mixed $responseCode
     */
    public function setResponseCode($responseCode)
    {
        $this->responseCode = $responseCode;
    }

    /**
     * @return mixed
     */
    public function getResponseText()
    {
        return $this->responseText;
    }

    /**
     * @param mixed $responseText
     */
    public function setResponseText($responseText)
    {
        $this->responseText = $responseText;
    }

    /**
     * @return mixed
     */
    public function getCisNote()
    {
        return $this->cisNote;
    }

    /**
     * @param mixed $cisNote
     */
    public function setCisNote($cisNote)
    {
        $this->cisNote = $cisNote;
    }

    /**
     * @return mixed
     */
    public function getRiskStatusResponseText()
    {
        return $this->riskStatusResponseText;
    }

    /**
     * @param mixed $riskStatusResponseText
     */
    public function setRiskStatusResponseText($riskStatusResponseText)
    {
        $this->riskStatusResponseText = $riskStatusResponseText;
    }

    /**
     * @return mixed
     */
    public function getRiskStatusRespondeCode()
    {
        return $this->riskStatusRespondeCode;
    }

    /**
     * @param mixed $riskStatusRespondeCode
     */
    public function setRiskStatusRespondeCode($riskStatusRespondeCode)
    {
        $this->riskStatusRespondeCode = $riskStatusRespondeCode;
    }

    /**
     * @return mixed
     */
    public function getRefundDateTime()
    {
        return $this->refundDateTime;
    }

    /**
     * @param mixed $refundDateTime
     */
    public function setRefundDateTime($refundDateTime)
    {
        $this->refundDateTime = $refundDateTime;
    }

    /**
     * @return mixed
     */
    public function getBillingReferences()
    {
        return $this->billingReferences;
    }

    /**
     * @param mixed $billingReferences
     */
    public function setBillingReferences($billingReferences)
    {
        $this->billingReferences = $billingReferences;
    }

    /**
     * @return mixed
     */
    public function getCustomerReference()
    {
        return $this->customerReference;
    }

    /**
     * @param mixed $customerReference
     */
    public function setCustomerReference($customerReference)
    {
        $this->customerReference = $customerReference;
    }


    public static function fromArray($data){
        if(!is_null($data)) {
            $refund = new RefundResponse();

            foreach ($data as $key => $value) {
                if (property_exists(get_class($refund), $key)) {
                    if ($key == "billingReferences") {
                        $refund->{$key} = BillingReferences::fromArray($value);
                    } elseif ($key == "customerReference") {
                        $refund->{$key} = CustomerReference::fromArray($value);
                    } else {
                        $refund->{$key} = $value;
                    }

                }
            }
            return $refund;
        }
    }

}

File: Payhub - SDK/java_SDK/PayHubSDK/src/main/java/com/payhub/ws/model/RefundResponse.java
/**
 * 
 */
package com.payhub.ws.model;

import com.fasterxml.jackson.annotation.JsonInclude;

/**
 * @author agustin
 *
 */
@JsonInclude(JsonInclude.Include.NON_NULL) 
public class RefundResponse {
	
	 
		public String saleTransactionId;
        
        public String refundTransactionId;
      
        public String token;
        
        public String approvalCode;
        
        public String processedDateTime;
        
        public String avsResultCode;
        
        public String verificationResultCode;
        
        public String batchId;
        
        public String responseCode;
        
        public String responseText;
        
        public String cisNote;
        
        public String riskStatusResponseText;
        
        public String riskStatusRespondeCode;
        
        public String refundDateTime;
        
        public String tokenizedCard;
        
        public BillingReferences billingReferences;

        public CustomerReference customerReference;

		/**
		 * @return the saleTransactionId
		 */
		public String getSaleTransactionId() {
			return saleTransactionId;
		}

		/**
		 * @param saleTransactionId the saleTransactionId to set
		 */
		public void setSaleTransactionId(String saleTransactionId) {
			this.saleTransactionId = saleTransactionId;
		}

		/**
		 * @return the refundTransactionId
		 */
		public String getRefundTransactionId() {
			return refundTransactionId;
		}

		/**
		 * @param refundTransactionId the refundTransactionId to set
		 */
		public void setRefundTransactionId(String refundTransactionId) {
			this.refundTransactionId = refundTransactionId;
		}

		/**
		 * @return the token
		 */
		public String getToken() {
			return token;
		}

		/**
		 * @param token the token to set
		 */
		public void setToken(String token) {
			this.token = token;
		}

		/**
		 * @return the approvalCode
		 */
		public String getApprovalCode() {
			return approvalCode;
		}

		/**
		 * @param approvalCode the approvalCode to set
		 */
		public void setApprovalCode(String approvalCode) {
			this.approvalCode = approvalCode;
		}

		/**
		 * @return the processedDateTime
		 */
		public String getProcessedDateTime() {
			return processedDateTime;
		}

		/**
		 * @param processedDateTime the processedDateTime to set
		 */
		public void setProcessedDateTime(String processedDateTime) {
			this.processedDateTime = processedDateTime;
		}

		/**
		 * @return the avsResultCode
		 */
		public String getAvsResultCode() {
			return avsResultCode;
		}

		/**
		 * @param avsResultCode the avsResultCode to set
		 */
		public void setAvsResultCode(String avsResultCode) {
			this.avsResultCode = avsResultCode;
		}

		/**
		 * @return the verificationResultCode
		 */
		public String getVerificationResultCode() {
			return verificationResultCode;
		}

		/**
		 * @param verificationResultCode the verificationResultCode to set
		 */
		public void setVerificationResultCode(String verificationResultCode) {
			this.verificationResultCode = verificationResultCode;
		}

		/**
		 * @return the batchId
		 */
		public String getBatchId() {
			return batchId;
		}

		/**
		 * @param batchId the batchId to set
		 */
		public void setBatchId(String batchId) {
			this.batchId = batchId;
		}

		/**
		 * @return the responseCode
		 */
		public String getResponseCode() {
			return responseCode;
		}

		/**
		 * @param responseCode the responseCode to set
		 */
		public void setResponseCode(String responseCode) {
			this.responseCode = responseCode;
		}

		/**
		 * @return the responseText
		 */
		public String getResponseText() {
			return responseText;
		}

		/**
		 * @param responseText the responseText to set
		 */
		public void setResponseText(String responseText) {
			this.responseText = responseText;
		}

		/**
		 * @return the cisNote
		 */
		public String getCisNote() {
			return cisNote;
		}

		/**
		 * @param cisNote the cisNote to set
		 */
		public void setCisNote(String cisNote) {
			this.cisNote = cisNote;
		}

		/**
		 * @return the riskStatusResponseText
		 */
		public String getRiskStatusResponseText() {
			return riskStatusResponseText;
		}

		/**
		 * @param riskStatusResponseText the riskStatusResponseText to set
		 */
		public void setRiskStatusResponseText(String riskStatusResponseText) {
			this.riskStatusResponseText = riskStatusResponseText;
		}

		/**
		 * @return the riskStatusRespondeCode
		 */
		public String getRiskStatusRespondeCode() {
			return riskStatusRespondeCode;
		}

		/**
		 * @param riskStatusRespondeCode the riskStatusRespondeCode to set
		 */
		public void setRiskStatusRespondeCode(String riskStatusRespondeCode) {
			this.riskStatusRespondeCode = riskStatusRespondeCode;
		}

		/**
		 * @return the refundDateTime
		 */
		public String getRefundDateTime() {
			return refundDateTime;
		}

		/**
		 * @param refundDateTime the refundDateTime to set
		 */
		public void setRefundDateTime(String refundDateTime) {
			this.refundDateTime = refundDateTime;
		}

		/**
		 * @return the tokenizedCard
		 */
		public String getTokenizedCard() {
			return tokenizedCard;
		}

		/**
		 * @param tokenizedCard the tokenizedCard to set
		 */
		public void setTokenizedCard(String tokenizedCard) {
			this.tokenizedCard = tokenizedCard;
		}

		/**
		 * @return the billingReferences
		 */
		public BillingReferences getBillingReferences() {
			return billingReferences;
		}

		/**
		 * @param billingReferences the billingReferences to set
		 */
		public void setBillingReferences(BillingReferences billingReferences) {
			this.billingReferences = billingReferences;
		}

		/**
		 * @return the customerReference
		 */
		public CustomerReference getCustomerReference() {
			return customerReference;
		}

		/**
		 * @param customerReference the customerReference to set
		 */
		public void setCustomerReference(CustomerReference customerReference) {
			this.customerReference = customerReference;
		}
        
        

}
